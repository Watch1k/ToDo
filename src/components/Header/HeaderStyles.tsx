import { IProps } from './Header'

export const getHeaderStyles = ({ darkMode }: IProps) => ({
  padding: 30,
  display: 'flex',
  justifyContent: 'center',
  background: darkMode ? '#2838ee' : 'transparent',
})

export const getTitleStyles = ({ darkMode }: IProps) => ({
  marginBottom: 0,
  color: darkMode ? '#fff' : 'inherit',
})
