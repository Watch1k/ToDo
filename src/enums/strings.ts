export enum Strings {
  header = 'ToDo List',
  darkMode = 'Dark Mode',
  mainCreateTusk = 'Create task',
  mainListOfTasks = 'List of tasks',
  results = 'Results',
  search = 'Search',
}
