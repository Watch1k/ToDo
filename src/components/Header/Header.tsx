import React from 'react'
import { Typography } from 'antd'
import { getHeaderStyles, getTitleStyles } from './HeaderStyles'
import { IHeaderProps } from '../../containers/HeaderContainer'

const { Title } = Typography

export type IProps = {
  title: string
} & IHeaderProps

const Header = (props: IProps) => {
  const { title } = props

  return (
    <header style={getHeaderStyles(props)} data-testid='header'>
      <Title style={getTitleStyles(props)}>{title}</Title>
    </header>
  )
}

export default Header
