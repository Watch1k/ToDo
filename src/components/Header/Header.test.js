import React from 'react'
import { render } from '@testing-library/react'
import Header from './'
import { Strings } from '../../enums/strings'

describe('Header component', () => {
  let wrapper
  const title = Strings.header

  beforeEach(() => {
    wrapper = render(<Header title={title} />)
  })

  it('Renders', () => {
    expect(wrapper.getByTestId('header')).toBeInTheDocument()
    expect(wrapper.getByText(title)).toBeInTheDocument()
  })

  it('Matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
