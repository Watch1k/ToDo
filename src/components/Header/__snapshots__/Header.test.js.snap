// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches snapshot 1`] = `
<DocumentFragment>
  <header
    data-testid="header"
    style="padding: 30px; display: flex; justify-content: center; background: transparent;"
  >
    <h1
      class="ant-typography"
      style="margin-bottom: 0px;"
    >
      ToDo List
    </h1>
  </header>
</DocumentFragment>
`;
